### PR TITLE
Add pair/only option for accelerator styles

### PR DESCRIPTION
**Pull request by akohlmey**
**Summary**

This is an alternate, more minimal approach to implement a pair/only option for suffixes that makes the (primary) suffix applicable only to pair styles, which can be useful for GPU acceleration to better overlap with host computations.

**Related Issue(s)**

This tries to implement the same behavior as #2512 but in a way that is not limited to KOKKOS.
GPU package should already be supported, but it could be also applied to USER-INTEL and USER-OMP.

**Author(s)**

Stan Moore, SNL and Axel Kohlmeyer, Temple U

**Licensing**

By submitting this pull request, I agree, that my contribution will be included in LAMMPS and redistributed under either the GNU General Public License version 2 (GPL v2) or the GNU Lesser General Public License version 2.1 (LGPL v2.1).

**Backward Compatibility**

No known issue.

**Post Submission Checklist**

- [x] The feature or features in this pull request is complete
- [x] Licensing information is complete
- [x] Corresponding author information is complete
- [x] The source code follows the LAMMPS formatting guidelines
- [x] Suitable new documentation files and/or updates to the existing docs are included
- [x] The added/updated documentation is integrated and tested with the documentation build system
- [x] The feature has been verified to work with the conventional build system
- [x] The feature has been verified to work with the CMake based build system
- [ ] Suitable tests have been added to the unittest tree.

### CODE REVIEW

**Comment by akohlmey** [comment]
@stanmoore1 here is a variant of the "pair/only" switch that can also be used with the GPU package and is a little less "disruptive". Please have a look, perhaps give it a try, and let me know what you think.

**Comment by stanmoore1** [comment]
@akohlmey nice, this looks good to me.

**Comment by akohlmey** [comment]
> @akohlmey nice, this looks good to me.

thanks. should we allow to revert the setting? like in https://github.com/lammps/lammps/pull/2533/commits/959f67962dcd1e3961afc0ece4236b3bc4292649 which still misses the reverting of the communication settings (which I currently don't know how to do). or should we just add something like this?

```
} else {
  if ((lmp->suffixp) && (me == 0))
    error->warning(FLERR,"Cannot turn 'pair/only' setting off after turning it on");
```

**Comment by stanmoore1** [comment]
Ah, there is logic in kokkos.cpp for reverting the communication setting for `CUDA/Aware`, I think that can just be reused for `pair/only`. I can add this.

**Comment by akohlmey** [comment]
> Ah, there is logic in kokkos.cpp for reverting the communication setting for `CUDA/Aware`, I think that can just be reused for `pair/only`. I can add this.

ok. assigning the PR to you to complete and test the KOKKOS part until you are happy with it. Will then look into the rest.

**Comment by stanmoore1** [comment]
@akohlmey I am testing this PR now

**Comment by akohlmey** [comment]
> @akohlmey I am testing this PR now

ok. i checked this morning and it works as expected for the GPU package and I also added documentation.

**Review by sjplimp — APPROVED** [review]

**Comment by ndtrung81** [comment]
@akohlmey Thanks for adding this option!

**Comment by akohlmey** [comment]
> @akohlmey Thanks for adding this option!

It was the idea of @stanmoore1 
I only re-implemented it slightly different so it can be applied to other packages as well.

**Review by ndtrung81 — APPROVED** [review]

**Comment by stanmoore1** [comment]
@akohlmey I've tested this for Kokkos, everything is working as far as I can tell. I think this is ready to merge from my POV.

**Comment by akohlmey** [comment]
> @akohlmey I've tested this for Kokkos, everything is working as far as I can tell. I think this is ready to merge from my POV.

excellent! I will have to wait until the CI testing clears and then merge it.

**Pull request merged**
